3-async-await/solution.js

```diff
@@ -19,17 +19,114 @@ Example:
 8. add any needed adjustment to solution() function
 9. as extra challenge: add Promise.race() and Promise.any(), and try to get the idea of what happens
 */
+const prices = require('./prices');
+const products = require('./products');
 
-function solution() {
+async function solution() {
     // YOUR SOLUTION GOES HERE
 
     // You generate your id value here
+    // NOTE : from what I tested, if you want to display the case where all 4 cases fail please assign 48 to id.
+    const id = randomId();
+    console.log(`The ID generated for this run is ${id}\n`);
+    
+
 
     // You use Promise.all() here
+    console.log('\tUsing Promise.all\n');
+    try {
+        const allPromises = await Promise.all([
+            prices(id),
+            products(id),
+        ]);
+        
+        console.log({ id, product: allPromises[1], price: allPromises[0] });
+
+    } catch (error) {
+        console.log('Entered Promise.all catch block');
+        console.error(error.message);
+    }
+    
+
 
     // You use Promise.allSettled() here
+    console.log('\n\tUsing Promise.allSettled\n');
+    try {
+        const promisesSettled = await Promise.allSettled([
+            prices(id),
+            products(id),
+        ]);
+        
+        const promisesFulfilled = promisesSettled.filter(promise => promise.status === 'fulfilled').map(result => result.value);
+        const promisesRejected = promisesSettled.filter(promise => promise.status === 'rejected').map(result => result.reason.message);
+        
+        if (promisesRejected.length) throw new Error(promisesRejected);
+
+        // Log the results, or errors, here
+        console.log({ id, product: promisesFulfilled[1], price: promisesFulfilled[0] });
+
+    } catch (error) {
+        /* NOTE : I've been trying to do proper error handling with all the promises but I had one struggle
```

Comment:
Keep in mind the difference between Promise.all and Promise.allSettled is that in the first an error could be thrown any moment so we should use a try/catch. But in the second case, it stores the results on each promise, so the error is stored too. It seems to me that line 64 is enough